src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -157,7 +160,7 @@ abstract public function getResults();
     /**
      * Get the relationship for eager loading.
      *
-     * @return \Illuminate\Database\Eloquent\Collection
+     * @return \Illuminate\Database\Eloquent\Collection<mixed, T>
```

Comment:
`mixed` should not be used as the key type for collections---it can only be a `int|string` (can also the `array-key` alias). However, I'd wager that most of these collection returning methods return a simple collection which would use `int` as the key type

```suggestion
     * @return \Illuminate\Database\Eloquent\Collection<int, T>
```

Or if it really can return `int` or `string` keys:
```suggestion
     * @return \Illuminate\Database\Eloquent\Collection<int|string, T>
```